marklogic-data-hub-central/ui-custom/src/components/TimelineVis/TimelineVis.tsx

```diff
@@ -0,0 +1,119 @@
+import React from "react";
+import "./TimelineVis.scss";
+import DateTime from "../DateTime/DateTime";
+import Value from "../Value/Value";
+import { getValByConfig } from "../../util/util";
+import _ from "lodash";
+import Timeline from "react-visjs-timeline";
+import "./TimelineVis.scss";
+import * as ReactDOMServer from 'react-dom/server';
+
+type Props = {
+    data?: any;
+    config?: any;
+};
+
+const COMPONENTS = {
+    DateTime: DateTime,
+    Value: Value,
+}
+
+/**
+ * Component for showing content on a timeline.
+ *
+ * @component
+ * @prop {object} config - Data table configuration object.
+ * @example
+ */
+const TimelineVis: React.FC<Props> = (props) => {
+
+    let data: any = [];
+    data = getValByConfig(props.data, props.config);
+    data = _.isNil(data) ? null : (Array.isArray(data) ? data : [data]);
+
+    let timelineData : any[]=[];
+
+    const getItems = (type) => (d) => {
+        let result = [];
+        result = props.config.popover.items.map((item, index) => {
+            if (!item) return null;
+            return (<div className="popover-row" key={"item-" + index}>
+                <span className="popover-label">{item.label}</span>
+                {(item.component) ?
+                    <span key={"item-" + index} className="popover-value">
+                        {React.createElement(COMPONENTS[item.component],
+                            { config: item.config, data: d, style: item.style, type: type}, null
+                        )}
+                    </span> :
+                    <span><span key={"item-" + index} className="popover-value">
+                        <Value data={d} config={item} getFirst={true} type={type}/>
+                    </span></span>
+                }
+            </div>)
+        })
+        return result;
+    }
+
+    data?.map((activity, id) => {
```

Comment:
All the direct references to the properties on the activity object here are hardcoded and need to be refactored. The customer could be displaying data from an "updates" parent with "update" children and the Timeline widget should work. The widget needs to pull from the data based on the paths values in the config. We can discuss.

Reply:
Done!!